### PR TITLE
Separated architecture docs from core concepts

### PR DESCRIPTION
**Pull request by LiraLemur**
Signed-off-by: Sara <lira.lemur@gmail.com>

### Description of the Change

Separated architectural component docs from core concepts, also added a diagram + fixed some minor errors + added some information from recent presentations. 
https://jira.hyperledger.org/browse/IR-492

Bonus: deleted throughput diagram from readme file (wasn't available anymore since waffle service was terminated). 

### Benefits

Better docs. Nicer looks

### Possible Drawbacks 

Do not see any

### CODE REVIEW

File: docs/source/core_concepts/glossary.rst [hunk 1]
```diff
@@ -45,142 +36,71 @@ Signable content is called payload, so the structure of a block looks like this:
...
 =======
 
-A command is an intention to change the `state <#world-state-view>`__.
-For example, in order to create a new `role <#role>`__ in Iroha you have to issue
-`Create role <../api/commands.html#create-role>`__ command.
-
-Consensus
-=========
-
-A consensus algorithm is a process in computer science used to achieve agreement
```
Comment (lebdron) [inline docs/source/core_concepts/glossary.rst]: I suggest to leave "consensus algorithm" and "Consensus, as an algorithm" description here, since it is general.

File: docs/source/maintenance/permissions.rst [hunk 2]
```diff
@@ -506,7 +506,7 @@ can_add_peer
 
 Allows adding `peers <../core_concepts/glossary.html#peer>`__ to the network.
 
-A new peer will be a valid participant in the next `consensus <../core_concepts/glossary.html#consensus>`__ round after an agreement on `transaction <../core_concepts/glossary.html#transaction>`__ containing "addPeer" `command <../core_concepts/glossary.html#command>`__.
```
Comment (lebdron) [inline docs/source/maintenance/permissions.rst]: The reference to consensus seems valid, maybe revert the change?

File: docs/source/architecture/index.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Is a part of `consensus <#consensus>`__.
+Adds missing blocks to `peers' <../core_concepts/glossary.html#peer>`__ chains (downloads them from other peers to preserve consistency).
+
+Ametsuchi Blockstore
+--------------------
+
+Iroha storage component, which stores blocks and a state generated from blocks, called `World State View <../core_concepts/glossary.html#world-state-view>`__.
+There is no way for the `client <../core_concepts/glossary.html#client>`__ to directly interact with Ametsuchi.
+
+
```
Comment (lebdron) [inline docs/source/architecture/index.rst]: Not sure if so many empty lines are required in the end of the file. 👀

File: docs/source/architecture/index.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+A consensus algorithm is a process in computer science used to achieve agreement on a single data value among distributed processes or systems.
+Consensus algorithms are designed to achieve reliability in a network involving multiple unreliable nodes.
+Solving that issue -- known as the consensus problem -- is important in distributed computing and multi-agent systems.
+
+*Consensus, as an algorithm*
+
+    An algorithm to achieve agreement on a block among peers in the network. By having it in the system, reliability is increased.
+
+*Consensus, as a component*
```
Comment (lebdron) [inline docs/source/architecture/index.rst]: Is should be possible to leave only "Consensus, as a component" section here.

File: docs/source/architecture/index.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+All the transactions that have not passed stateful validation will be dropped and not included in a verified proposal.
+
+Block Creator
+-------------
+
+System component that forms a block from a set of transactions that have passed `stateless <../core_concepts/glossary.html#stateless-validation>`__ and `stateful <../core_concepts/glossary.html#stateful-validation>`__ validation for further propagation to `consensus <#consensus>`__.
+
+Block creator, together with the `Verified Proposal Creator <#verified-proposal-creator>`_ form a component called `Simulator <https://github.com/hyperledger/iroha/tree/master/irohad/simulator>`_.
+
+Consensus
```
Comment (lebdron) [inline docs/source/architecture/index.rst]: And rename the section to "Block Consensus (YAC)" to avoid confusion with the corresponding section in core concepts.

File: docs/source/maintenance/permissions.rst (hunk 2 above)
Comment (LiraLemur) [reply docs/source/maintenance/permissions.rst]: Tbh, I did not change this part manually - it is an automated part of the doc. 
I believe it was lost because I'd deleted the section. I'll check this after I bring the section back.

File: docs/source/core_concepts/glossary.rst [hunk 6]
```diff
@@ -327,15 +225,6 @@ Stateful Validation
...
-=========================
-
-Internal Iroha component that performs `stateful validation <#stateful-validation>`_
-of `transactions <#transaction>`__ contained in received `proposal <#proposal>`__.
-On the basis of transactions that have been passed stateful validation **verified proposal**
-will be created and passed to `Block Creator <#block-creator>`__.
-All the transactions that have not passed stateful validation will be dropped
-and not included in a verified proposal.
 
 World State View
```
[the remark this answers is not in the document]
Comment (LiraLemur) [reply docs/source/core_concepts/glossary.rst]: ok, will do

File: docs/source/architecture/index.rst (hunk 5 above)
Comment (LiraLemur) [reply docs/source/architecture/index.rst]: done!

File: docs/source/architecture/index.rst [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+Architecture
+============
+
+.. toctree::
+   :maxdepth: 2
+
+HL Iroha network consists of several essential components that provide the communication between the nodes. You can learn about them below. 
+
+.. image:: ../../image_assets/PD.png
```
[the remark this answers is not in the document]
Comment (LiraLemur) [reply docs/source/architecture/index.rst]: done! Will commit soon

File: docs/source/maintenance/permissions.rst (hunk 2 above)
Comment (LiraLemur) [reply docs/source/maintenance/permissions.rst]: The thingy seems to have fixed itself

File: docs/source/architecture/index.rst (hunk 3 above)
Comment (LiraLemur) [reply docs/source/architecture/index.rst]: 🤦

File: docs/source/architecture/index.rst (hunk 4 above)
Comment (LiraLemur) [reply docs/source/architecture/index.rst]: Just left it as "consensus". With an additional reference link

**Comment by LiraLemur** [comment]
(Hopefully) fixed everything @lebdron mentioned here.
